src/wallet/rpcwallet.cpp

```diff
@@ -3169,6 +3169,76 @@ UniValue generate(const JSONRPCRequest& request)
     return generateBlocks(coinbase_script, num_generate, max_tries, true);
 }
 
+UniValue rescanblockchain(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() > 2) {
+        throw std::runtime_error(
+            "rescanblockchain (\"start_height\") (\"stop_height\")\n"
+            "\nRescan the local blockchain for wallet related transactions.\n"
+            "\nArguments:\n"
+            "1. \"start_height\"    (number, optional) block height where the rescan should start\n"
+            "2. \"stop_height\"     (number, optional) block height where the rescan should stop\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"start_height\"     (number) The block height where the rescan has started. If omitted, rescan starts from the genesis block.\n"
+            "  \"stop_height\"      (number) The height of the last rescanned block. If omitted, rescan stops at the chain tip.\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("rescanblockchain", "\"100000 120000\"")
+            + HelpExampleRpc("rescanblockchain", "\"100000 120000\"")
+            );
+    }
+
+    LOCK2(cs_main, pwallet->cs_wallet);
+
+    CBlockIndex *pindexStart = chainActive.Genesis();
+    CBlockIndex *pindexStop = nullptr;
+    if (!request.params[0].isNull()) {
+        pindexStart = chainActive[request.params[0].get_int()];
+        if (!pindexStart) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid start_height");
+        }
+    }
+
+    if (!request.params[1].isNull()) {
+        pindexStop = chainActive[request.params[1].get_int()];
+        if (!pindexStop || pindexStop->nHeight < pindexStart->nHeight) {
```

Comment:
I think with this (and with the above pindexStart check), I'd like to see these errors differentiate between:

1. Invalid Start Height: no negative heights
2. Invalid Start Height: Beyond what's been synced
3. Invalid Stop Height: No negative heights
4. Invalid Stop Height: Beyond what's been synced
5. Invalid Range: stop must be greater than start.

At the very least, 5. should be represented because it could be the start_height which is invalid.

Reply:
Added case 5.